### ISSUE TITLE
[Question]Upgrade from 2.0.x to 2.1 property name and sql parse problems

### ISSUE THREAD

**Issue by vtliem** [orientechnologies/orientdb#4768]
My webapp is using OrientDB (1.7..2.0).
DB is created by JAVA API (not using SQL command).
Now, I upgraded to 2.1 and found some code are broken.
(DB created with OrientDB 2.1)

1 . ` Oclass.createProperty(name,type) ` is not allowed `reserved keywords`. 
-> I read OrientDB code and changed `Oclass.createProperty` to `OClassImpl.addPropertyInternal`. Is it  safe? How I can add a property with  `reserved keywords`?
2 .  Add `property` which contains `-` character, orientDb raised `Sql parse exception` although it is seem valid property name. If `iCheckExistendRecords` parameter is `false`, the `exception` is not raised and I can add `property`. This is a bug or `-` is an invalid chararer?
3.  Query
```sql
SELECT `from` FROM message
```
OR
```sql
SELECT [from] FROM message
```
OR
```sql
SELECT user-agent FROM session
```
OR
```sql
SELECT `user-agent` FROM session
```
-->raise `Sql parse exception`
```sql
SELECT "user-agent" FROM session
```
returned 
```
{"@rid":"#-1:0","user-agent":"user-agent"}
```

How to escape this `keyword`?

4.  `OSimpleKeyIndexDefinition` now  required `version` parameter. What is this?

**Comment by lvca** [comment]
This is something already addressed by @luigidellaquila. I think today this could be fixed in our 2.1.1-SNAPSHOT. @luigidellaquila any news?

**Comment by luigidellaquila** [comment]
Hi @vtliem 

in v 2.1 you can escape keywords and special characters like '-' using backtick, eg. this is correct:

```sql
SELECT `user-agent` FROM session
```

In 2.0 final the - character did not work fine, I fixed it a couple of days ago, you can find the fix in branch "2.1.x". The fix will be released with 2.1.1 hotfix release
